Tools/Create-Azure-Sentinel-Solution/common/createCCPConnector.ps1

```diff
@@ -0,0 +1,880 @@
+                    elseif ($armResource.properties.auth.type.ToLower() -eq 'apikey') {
+                        # ApiKey 
+                        $hasApiKey = [bool]($armResource.properties.auth.PSobject.Properties.name -match "ApiKey")
+                        if ($hasApiKey) {
+                            $apiKeyProperty = $armResource.properties.auth.ApiKey
+                            $placeHoldersMatched = $apiKeyProperty | Select-String $placeHolderPatternMatches -AllMatches
+    
+                            if ($placeHoldersMatched.Matches.Value.Count -gt 0) {
+                                $placeHolderName = $placeHoldersMatched.Matches.Value.replace("{{", "").replace("}}", "")
+                                $armResource.properties.auth.ApiKey = "[[parameters('apikey')]"
+    
+                                $templateContentConnections.properties.mainTemplate = addNewParameter -templateResourceObj $templateContentConnections.properties.mainTemplate -parameterName 'apikey' -isSecret $true
+                            }
+                        }
+                    }
+    
+                    if ($armResource.properties.request.apiEndPoint.contains("{{")) {
+                        # identify any placeholders in apiEndpoint
+                        $endPointUrl = $armResource.properties.request.apiEndPoint                    
+                        $placeHoldersMatched = $endPointUrl | Select-String $placeHolderPatternMatches -AllMatches
+                        
+                        if ($placeHoldersMatched.Matches.Value.Count -gt 0) {
+                            # has some placeholders 
+                            $finalizedEndpointUrl = ""
+                            $finalizedEndpointUrl = "[[concat("
+                            $closureBrackets = ")]"
+    
+                            foreach ($currentPlaceHolder in $placeHoldersMatched.Matches.Value) {
+                                $placeHolderName = $currentPlaceHolder.replace("{{", "").replace("}}", "")
+                                $splitEndpoint = $endPointUrl -split "($currentPlaceHolder)"
+                                $commaCount = 0
+                                foreach($splitItem in $splitEndpoint) {
+                                    if ($splitItem -eq $currentPlaceHolder) {
+                                        if ($finalizedEndpointUrl -eq '') {
+                                            $finalizedEndpointUrl += "parameters('" + $placeHolderName + "')"
+                                        } else {
+                                            $finalizedEndpointUrl += ", parameters('" + $placeHolderName + "')"
+                                        }
+                                        
+                                        $templateContentConnections.properties.mainTemplate = addNewParameter -templateResourceObj $templateContentConnections.properties.mainTemplate -parameterName "$placeHolderName" -isSecret $false
```

Comment:
check for secret sensitivity here in the code

Reply:
updated